ArduCopter/land_detector.cpp

```diff
@@ -174,10 +190,16 @@ void Copter::update_throttle_thr_mix()
         // check for requested decent
         bool descent_not_demanded = pos_control.get_desired_velocity().z >= 0.0f;
 
-        if ( large_angle_request || large_angle_error || accel_moving || descent_not_demanded) {
-            attitude_control.set_throttle_mix_max();
-        } else {
+        bool xy_speed_low = (position_ok() || optflow_position_ok()) && xy_speed_cms <= 125.0f;
+        bool xy_speed_demand_low = pos_control.is_active_xy() && xy_des_speed_cms <= 125.0f;
+        bool slow_descent_demanded = pos_control.is_active_z() && des_climb_rate_cms < 0.0f && des_climb_rate_cms >= -100.0f;
+
+        bool slow_auto_descent = xy_speed_demand_low && xy_speed_low && slow_descent_demanded;
+
+        if ( slow_auto_descent || !(large_angle_request || large_angle_error || accel_moving || descent_not_demanded) ) {
```

Comment:
The original logic was:
```
if ( large_angle_request || large_angle_error || accel_moving || descent_not_demanded) {
    attitude_control.set_throttle_mix_max();
} else {
    attitude_control.set_throttle_mix_min();
}
```

Let original_criteria = large_angle_request || large_angle_error || accel_moving || descent_not_demanded

```
if (original_criteria) {
    attitude_control.set_throttle_mix_max();
} else {
    attitude_control.set_throttle_mix_min();
}
```

Invert if statement:
```
if (!original_criteria) {
    attitude_control.set_throttle_mix_min();
} else {
    attitude_control.set_throttle_mix_max();
}
```

Add new criteria:
```
if (slow_auto_descent || !original_criteria) {
    attitude_control.set_throttle_mix_min();
} else {
    attitude_control.set_throttle_mix_max();
}
```

Sub out original_criteria:
```
if (slow_auto_descent || !(large_angle_request || large_angle_error || accel_moving || descent_not_demanded)) {
    attitude_control.set_throttle_mix_min();
} else {
    attitude_control.set_throttle_mix_max();
}
```